packages/runtime/src/useCases/transport/files/CreateTokenForFile.ts

```diff
@@ -1,22 +1,57 @@
 import { Result } from "@js-soft/ts-utils";
 import { CoreAddress, CoreDate, CoreId } from "@nmshd/core-types";
-import { AccountController, File, FileController, TokenContentFile, TokenController } from "@nmshd/transport";
+import { AccountController, File, FileController, PasswordProtectionCreationParameters, TokenContentFile, TokenController } from "@nmshd/transport";
 import { Inject } from "@nmshd/typescript-ioc";
+import { DateTime } from "luxon";
+import { nameof } from "ts-simple-nameof";
 import { TokenDTO } from "../../../types";
-import { AddressString, FileIdString, ISO8601DateTimeString, RuntimeErrors, SchemaRepository, SchemaValidator, UseCase } from "../../common";
+import {
+    AddressString,
+    FileIdString,
+    GenericInputValidator,
+    ISO8601DateTimeString,
+    RuntimeErrors,
+    SchemaRepository,
+    UseCase,
+    ValidationFailure,
+    ValidationResult
+} from "../../common";
 import { TokenMapper } from "../tokens/TokenMapper";
 
 export interface CreateTokenForFileRequest {
     fileId: FileIdString;
     expiresAt?: ISO8601DateTimeString;
     ephemeral?: boolean;
     forIdentity?: AddressString;
+    passwordProtection?: {
+        /**
+         * @minLength 1
+         */
+        password: string;
+        passwordIsPin?: true;
+    };
 }
 
-class Validator extends SchemaValidator<CreateTokenForFileRequest> {
+class Validator extends GenericInputValidator<CreateTokenForFileRequest> {
     public constructor(@Inject schemaRepository: SchemaRepository) {
         super(schemaRepository.getSchema("CreateTokenForFileRequest"));
     }
+
+    public override validate(input: CreateTokenForFileRequest): ValidationResult {
+        const validationResult = super.validate(input);
+        if (!validationResult.isValid()) return validationResult;
+
+        if (input.expiresAt && DateTime.fromISO(input.expiresAt) <= DateTime.utc()) {
+            validationResult.addFailure(
+                new ValidationFailure(
+                    RuntimeErrors.general.invalidPropertyValue(`'${nameof<CreateTokenForFileRequest>((r) => r.expiresAt)}' must be in the future`),
+                    nameof<CreateTokenForFileRequest>((r) => r.expiresAt)
+                )
+            );
+        }
```

Comment:
so this is perfectly fine - but does not belong in this PR.

Reply:
as it's done already keep it.

Reply:
Hm. Now I see another issue with the implementation - which is **exactly** the reason why this should be in another PR

1. use CoreDate instead of DateTime
2. use `CoreDate#isExpired()` to check against `DateTime.now()`